dpp/src/main/java/org/dashj/platform/dashpay/BlockchainIdentity.kt

```
@@ -484,6 +485,19 @@ class BlockchainIdentity {
         finalizeIdentityRegistration(assetLockTransaction!!)
     }
 
+    private fun determineBlockHeight(assetLockTransaction: AssetLockTransaction?): Long =
+        if (assetLockTransaction!!.confidence.confidenceType == TransactionConfidence.ConfidenceType.BUILDING) {
+            assetLockTransaction!!.confidence.appearedAtChainHeight
+        } else {
+            try {
+                // determine the height from Platform Core (getTransaction)
+                val response = platform.client.getTransactionKotlin(assetLockTransaction!!.txId.toString())
+                response?.height ?: DashSystem.get(wallet!!.params).blockChain.bestChainHeight
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)